From adecbb8cf8588bee92b639bdd3fa05007266a5d2 Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Tue, 16 Aug 2022 14:34:12 +0300
Subject: [PATCH] [SW-95918] gaudi2: security: set spi_dma security level to privilege

Recently SPI DMA was used at preboot stage for image validation from
flash (RAS requirement). This caused PSOC_AXI_ERR_RESP when security is
enabled as SRAM is closed by privilege RR by that stage and SPI DMA is
trying to copy to SRAM for validation. This is due to the security level
of spi_dma being a non-privilege level by default.

SPI DMA is used only by FW and copies data from SPI to various locations
including those accessible only by privilege entities. Also, all the
SPI DMA registers are privilege registers. So, it makes sense to assign
the SPI_DMA security level to privilege.

Hence, this patch sets the spi dma security level to privilege.

Change-Id: Ic393254b2b4605fa0604fb4a7fff0eb9deaf91e8
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index c915a64..d48dbfb 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -781,6 +781,13 @@
 	regmap_writel(HL_SEC_AXPROT_OVRD_SEC, mmPCIE_ELBI_RR_MSTR_IF_CORE_LBW_AXPROT_OVRD);
 }
 
+#define SPI_DMA_AWPROT_PRIV		(1)
+static void ppboot_secure_spi_dma(void)
+{
+	/* make SPI_DMA as priv entity */
+	regmap_writel(SPI_DMA_AWPROT_PRIV, mmPSOC_GLOBAL_CONF_SPI_DMA_AWPROT);
+}
+
 static void preboot_secure_pcie(void)
 {
 	regmap_writel((PCIE_WRAP_PCIE_PROT_OVR_RD_EN_MASK |
@@ -1211,6 +1218,7 @@
 {
 	ppboot_secure_elbi();
 	ppboot_secure_hbm();
+	ppboot_secure_spi_dma();
 
 	if (!preboot_security_is_enabled()) {
 		/* Enable JTAG to override disablement in arcboot */
